src/commands/dev.ts

```diff
@@ -877,63 +928,16 @@ export async function command(commandOptions: CommandOptions) {
     }
   }
 
-  onProcessExit(() => {
-    hmrEngine.disconnectAllClients();
-  });
-
-  console.log = (...args) => {
-    messageBus.emit('CONSOLE', {level: 'log', args});
-  };
-  console.warn = (...args) => {
-    messageBus.emit('CONSOLE', {level: 'warn', args});
-  };
-  console.error = (...args) => {
-    messageBus.emit('CONSOLE', {level: 'error', args});
-  };
-
-  const ips = Object.values(os.networkInterfaces())
-    .reduce((every: os.NetworkInterfaceInfo[], i) => [...every, ...(i || [])], [])
-    .filter((i) => i.family === 'IPv4' && i.internal === false)
-    .map((i) => i.address);
-
+  // Announce server has started
   const protocol = config.devOptions.secure ? 'https:' : 'http:';
-
-  paint(messageBus, config.scripts, undefined, {
-    protocol,
+  messageBus.emit('SERVER_START', {
```

Comment:
This is a new event that repaints. This is called when we have everything ready to go, including server start time, protocol, port, etc.—all the things that were originally required by `paint()`.